src/constructors/regular.c

```diff
@@ -134,6 +134,145 @@ igraph_error_t igraph_star(igraph_t *graph, igraph_integer_t n, igraph_star_mode
     return IGRAPH_SUCCESS;
 }
 
+/**
+ * \ingroup generators
+ * \function igraph_wheel
+ * \brief Creates a \em wheel graph, a union of a star graph and a
+ *        cycle graph. Note that in case of a graph with 2 vertices,
+ *        the non-center vertex will have a self loop (may imagine
+ *        the earth is the center of the wheel, the moon and its orbit
+ *        is the rim of the wheel).
+ * \param graph Pointer to an uninitialized graph object, this will
+ *        be the result.
+ * \param n Integer constant, the number of vertices in the graph.
+ * \param mode Constant, gives the type of the star graph to
+ *        create. Possible values:
+ *        \clist
+ *        \cli IGRAPH_WHEEL_OUT
+ *          directed wheel graph, edges point
+ *          \em from the center to the other vertices.
+ *        \cli IGRAPH_WHEEL_IN
+ *          directed wheel graph, edges point
+ *          \em to the center from the other vertices.
+ *        \cli IGRAPH_WHEEL_MUTUAL
+ *          directed wheel graph with mutual edges.
+ *        \cli IGRAPH_WHEEL_UNDIRECTED
+ *          an undirected wheel graph is
+ *          created.
+ *        \endclist
+ * \param center Id of the vertex which will be the center of the
+ *          graph.
+ * \return Error code:
+ *         \clist
+ *         \cli IGRAPH_EINVVID
+ *           invalid number of vertices.
+ *         \cli IGRAPH_EINVAL
+ *           invalid center vertex.
+ *         \cli IGRAPH_EINVMODE
+ *           invalid mode argument.
+ *         \endclist
+ *
+ * Time complexity: O(|V|), the
+ * number of vertices in the graph.
+ *
+ * \sa \ref igraph_lattice(), \ref igraph_ring(), \ref igraph_star, \ref igraph_tree()
+ * for creating other regular structures.
+ *
+ */
+
+igraph_error_t igraph_wheel(igraph_t *graph, igraph_integer_t n, igraph_wheel_mode_t mode,
+                igraph_integer_t center) {
+
+    igraph_integer_t  v;
+    igraph_star_mode_t star_mode;
+    igraph_vector_int_t rim_edges = IGRAPH_VECTOR_NULL;
+    igraph_integer_t i;
+
+    /* create a star and also make use of its existing pre-check */
+    /* works for all input parameters                            */
+
+    switch(mode)
```

Comment:
Please leave a space after keywords such as `switch`, `if`, `for`, etc.

Reply:
Updated, thanks for reminding the style standard.